Add Optional Expiry to Auth Tokens (#844)
hashicorp/terraform-provider-tfe: website/docs/r/organization_token.html.markdown

* `force_regenerate` - (Optional) If set to `true`, a new token will be
  generated even if a token already exists. This will invalidate the existing
  token!
* `expired_at` - (Optional) A date and time in which the organization token will expire. The expiration date must be passed in
  iso8601 format. If no expiration date is supplied, the expiration date will default to null and never expire.

## Attributes Reference

* `id` - The ID of the token.
* `token` - The generated token.

> * `expired_at` - The token's expiration date. The expiration date must be passed in
> iso8601 format. If no expiration date is supplied, the expiration date will default to null and never expire.

Reviewer: I'm so sorry I gave you the wrong feedback. I was looking at a single commit and not the entire change. I didn't see this was already added to the section above and I got them mixed up.

```suggestion
```

> * `expired_at` - The token's expiration date. The expiration date must be passed in
> iso8601 format. If no expiration date is supplied, the expiration date will default to null and never expire.

Reviewer: ```suggestion
* `expired_at` - The token's expiration date. The expiration date must be a date/time string in 
RFC3339 format (For example, "2024-12-31T23:59:59Z"). If no expiration date is supplied, the expiration date will default to null and never expire.
```

* `force_regenerate` - (Optional) If set to `true`, a new token will be
  generated even if a token already exists. This will invalidate the existing
  token!

> * `expired_at` - (Optional) The token's expiration date. The expiration date must be a date/time string in RFC3339 

Author: I've updated the description for `expired_at` per your suggestions, but opted to leave the "optional" here since the expiry is optional and seems consistent with the other optional arguments, but will happily remove it if preferred! Additionally, I moved the example usage back to where it initially was (beneath the arguments docs!) to keep it more inline with the expiry for context. 

Let me know what you think, @brandonc! 😄

format (e.g., "2024-12-31T23:59:59Z"). If no expiration date is supplied, the expiration date will default to null and 
never expire.

## Example Usage

When a token has an expiry:

```hcl

> resource "tfe_organization_token" "test" {
>   organization = "my-org-name"
> }
> 

Reviewer: I think this is an extra resource defined in the example. If you are trying to show examples with and without an expired_at, that would be OK, but they should be separated by different code blocks maybe

```suggestion
```

Author: Whoops! That was an extra resource 🙈

resource "time_rotating" "example" {
  rotation_days = 30
}

resource "tfe_organization_token" "test" {
  organization = data.tfe_organization.org.name

>   expired_at = time_rotating.example.id

Reviewer: OK so I tested this out and the docs for the provider seem to be wrong. `id` returns the base timstamp and `rotation_rfc3339` returns the offset timestamp. The examples should look like this:

```suggestion
  expired_at = time_rotating.example.rotation_rfc3339
```

Reviewer: Fix here https://github.com/hashicorp/terraform-provider-time/pull/137

Author: Good to know, thank you! I've updated the examples

* `force_regenerate` - (Optional) If set to `true`, a new token will be
  generated even if a token already exists. This will invalidate the existing
  token!

> * `expired_at` - (Optional) A date and time in which the organization token will expire.

Reviewer 2: I think that this would be a good place to include information on what format we expect this value to be passed in, maybe with examples. For example, we should specify that this value should be UTC https://hashicorp.atlassian.net/browse/TF-5300

I think it would also be worth mentioning that if this is not set that the token will never expire

```hcl
resource "tfe_organization_token" "test" {
  organization = "my-org-name"
}

resource "time_rotating" "example" {
  rotation_days = 30
}

resource "tfe_organization_token" "test" {
  organization = data.tfe_organization.org.name
  expired_at = time_rotating.example.id
}
```

Generating the `expired_at` string using the date tool in unix systems (darwin):
```
date -Iseconds -v"+30d"
```

Generating the `expired_at` string using the date tool in unix systems (linux):
```
date -Iseconds -d"+30 days"
```

Generating the `expired_at` string using the `timeadd` Terraform function:
```
$ terraform console
> timeadd(timestamp(), "720h")

> "2023-07-21T02:02:23Z"

Reviewer: With the time_rotating example, I don't think all this detail is necessary anymore. Also, can you add the `expired_at` token below in the "Attributes Reference" with specific detail/example of the format? (both resources)

Author: Good call out, will do!
